src/UploadValidator/Base64UploadValidator.php

```diff
@@ -0,0 +1,19 @@
+<?php
+namespace Josegonzalez\Upload\UploadValidator;
+
+use Josegonzalez\Upload\UploadValidator\DefaultUploadValidator;
+
+class Base64UploadValidator extends DefaultUploadValidator
+{
+
+    /**
+     * Check's data for any upload errors.
+     * pairs, where the path on disk maps to name which each file should have.
+     *
+     * @return bool `true` if upload failed
+     */
+    public function hasUploadFailed()
+    {
+        return !base64_decode($this->entity->get($this->field), true);
```

Comment:
How can one do other validations like for e.g checking for particular mimetype / file type?

Reply:
I don't think is possible to do check on mimetype / file type without having the file already written on the disk

Reply:
Wouldn't it be possible having validations using the example code provided in this comment https://github.com/FriendsOfCake/cakephp-upload/pull/495#issuecomment-412262073 ?

Reply:
I just tried and it works. I wasn't aware of such method to determine the mimetype.